src/mpi/datatype/typerep/dataloop/dataloop.c

```diff
@@ -397,7 +397,7 @@ void MPII_Dataloop_alloc_and_copy(int kind,
     new_loop_sz += loop_sz + off_sz + blk_sz + ptr_sz + extent_sz + old_loop_sz;
 
     /* allocate space */
-    new_loop = (MPII_Dataloop *) MPL_malloc(new_loop_sz, MPL_MEM_DATATYPE);
+    new_loop = (MPII_Dataloop *) MPL_calloc(1, new_loop_sz, MPL_MEM_DATATYPE);
```

Comment:
So the size of this (c)alloc is sort of unbounded depending on the datatype. E.g. an indexed type with many elements will create a very large dataloop structure. A more targeted approached to fixing the uninitialized problem is probably better if we can find it.

Reply:
After offline discussion we decided to just merge this as-is. 1. this is ch3:sock so there are limited users. 2. if a dataloop consists of large arrays, there is already a large cost to copy the arrays to the dataloop structure so the extra cost of zero initialization is unlikely to be significant.